azure-iot-device/azure/iot/device/common/transport_exceptions.py

```diff
@@ -0,0 +1,40 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module defines errors that may be raised from a transport"""
+
+from .chainable_exception import ChainableException
+
+
+class ConnectionFailedError(ChainableException):
+    """
+    Connection failed to be established
+    """
+
+    pass
+
+
+class ConnectionDroppedError(ChainableException):
+    """
+    Previously established connection was dropped
+    """
+
+    pass
+
+
+class UnauthorizedError(ChainableException):
```

Comment:
>UnauthorizedError [](start = 6, length = 17)

Might be somewhat confusing that this shares a name with a service error, even though it's in a different namespace.